telosnetwork/telos-wallet
# #658 | ERC1155 transfers and owners

**blocked by #599**

# Fixes #658
# Fixes #661 

## Description
This PR does a few things:
1. enables transfers for ERC1155 NFTs via the NFT detail page
2. adds an Owners tab to the NFT detail page for ERC1155 NFTs
3. fixes the back navigation issue on the NFT detail page
4. makes a11y improvements to the `CurrencyInput` component

## Test scenarios
- go to https://deploy-preview-662--wallet-develop-mainnet.netlify.app/
- log in with an account which has both ERC1155 (at least 2 of a single tokenID) and ERC721 tokens
- go to the NFT inventory page
- change the rows per page option to anything besides the default
- click on an ERC721 token
    - there should be no `Owners` tab on the page
- click the site Back button 
    - you should be taken back to the Inventory page
    - the query params in the browser address bar should still contain the rows per page you changed to
- go back to the ERC721 NFT detail page
- click on the `Transfer` tab
    - you should see no 'Quantity' field
- transfer the NFT to another account you own
    - the transfer should still work
- go back to the NFT Inventory page
- click on an ERC1155 NFT
    - there should be an `Owners` tab
- click on `Owners` tab
    - you should see a list of token holders
- click on a token holder
    - you should be taken to the teloscan page for that address
- go back to the detail page for the ERC1155 NFT
- test the Owners search field
    - the list of owners should filter based on input
- click on the `Transfer` tab
    - you should see a 'Quantity' field
- play around with the quantity field
    - it should behave reasonably (error state on blur if invalid amount, error state if you enter too high a number, error state if you enter 0)
    - as you experiment, the state of the Transfer Collectible button should reflect the validity of what you enter, i.e. be disabled if you don't enter a valid amount
- transfer 1 NFT to an account you own
    - the amount owned by you should reduce by 1
- switch to that other account you sent the NFT to
    - the amount owned by that account should be correct (i.e. 1)
 - send the 1 NFT from your second account back to your first
     - the amount owned by you should reduce to 0 and the transfer tab should disappear
 - switch back to your first account
 - transfer all of the NFT to you second account _by manually typing in the max amount_
     - the amount owned by you should reduce to 0 and the transfer tab should disappear
 - switch back to your second account
     - you should see owned by you reflect the amount you just sent
 - transfer all of the NFT to you first account _by clicking the Total text above the input_
     - the amount owned by you should reduce to 0 and the transfer tab should disappear
 - switch back to your first account
     - you should see owned by you reflect the amount you just sent

## Checklist:
